integration_test/sql/migration.exs

```diff
@@ -101,6 +101,75 @@ defmodule Ecto.Integration.MigrationTest do
     end
   end
 
+  defmodule MigrationTestTable do
+    use Ecto.Model
+
+    schema "migration_test_table" do
+      field :to_be_modified, :string
+      field :to_be_removed, :integer
+    end
+  end
+
+  defmodule AlterColumnMigration do
+    use Ecto.Migration
+
+    def up do
+      create table(:migration_test_table) do
+        add :to_be_modified, :integer
+        add :to_be_removed, :integer
+      end
+
+      alter table(:migration_test_table) do
+        modify :to_be_modified, :string
+      end
+
+      execute "INSERT INTO migration_test_table (to_be_modified) VALUES ('foo')"
+    end
+
+    def down do
+      drop table(:migration_test_table)
+    end
+  end
+
+  defmodule DropColumnMigration do
+    use Ecto.Migration
+
+    def up do
+      create table(:migration_test_table) do
+        add :to_be_modified, :string
+        add :to_be_removed, :integer
+      end
+
+      alter table(:migration_test_table) do
+        remove :to_be_removed
+      end
+
+      execute "INSERT INTO migration_test_table (to_be_modified) VALUES ('foo')"
+    end
+
+    def down do
+      drop table(:migration_test_table)
+    end
+  end
+
+  @tag :modify_column
+  test "modify column" do
+    import Ecto.Query, only: [from: 2]
+    assert :ok == up(TestRepo, 20080906120000, AlterColumnMigration, log: false)
+    assert "foo" == TestRepo.one from p in MigrationTestTable, select: p.to_be_modified
+    :ok = down(TestRepo, 20080906120000, AlterColumnMigration, log: false)
+  end
+
+  @tag :remove_column
+  test "remove column" do
+    import Ecto.Query, only: [from: 2]
+    assert :ok == up(TestRepo, 20090906120000, DropColumnMigration, log: false)
+    assert catch_error(TestRepo.one from p in MigrationTestTable, select: p.to_be_removed)
+    unless TestRepo.adapter == Ecto.Adapters.Postgres do
+      :ok = down(TestRepo, 20090906120000, DropColumnMigration, log: false)
+    end
```

Comment:
Since the `up/4` migration will implicitly commit the MySQL test transaction, the `down/4` migration will clean-up after the test by dropping the test table.  This is only necessary for MySQL -- for PostgreSQL the `down/4` call is unnecessary because rolling back the test transaction will drop the table anyway.

The test asserts that the `to_be_removed` column was removed by looking for an error when trying to access the non-existent row.  The `down/4` call is wrapped in an if-check because PostgreSQL does not allow any more statements to be made within a transaction once an error has been thrown.  Thus, the test table cannot be dropped by the Postgres adapter.

I do not like calling out the Postgres adapter specifically in a test that is meant to be generic, but I couldn't think of a better way to get around the PostgreSQL error.  Is there a better, more generic way of cleaning up the test for MySQL without running into an error with PostgreSQL?  The assertion also relies on an error being raised.  Is there a generic Ecto way to check the absence of a column in a table?  If there was, that would remove the need for the if-check.